Use a global Stack cache on macOS

ttuegel opened the pull request:
Set HOMEBREW_NO_ENV_FILTERING=1 so that we can point Stack at the global cache
in the Anka template. Otherwise, Stack-inside-Homebrew will download and compile
all dependencies on every CI job.

See also: #1813

cos commented:
@laurayuwen, is this ready for merging?

laurayuwen commented:
@cos Yes.

cos commented on k-distribution/tutorial/1_k/5_types/lesson_9/lambda.k:
```diff
@@ -46,18 +58,18 @@ module LAMBDA
 
   syntax KItem ::= tenv(Map)
-  rule <k> T:Type ~> (tenv(TEnv) => .) ...</k>  <tenv> _ => TEnv </tenv>
+  rule <k> `T:Type ~> tenv(TEnv) => T` ...</k> <tenv> `_ => TEnv` </tenv>
```
should work without the second set of backticks

laurayuwen replied:
@dwightguth @cos all backticks removed

cos commented:
Jenkins, test this please

cos commented:
Jenkins, test this please

ttuegel committed:
Jenkinsfile: Set HOMEBREW_NO_ENV_FILTERING when building bottle

Set HOMEBREW_NO_ENV_FILTERING=1 so that we can point Stack at the global cache
in the Anka template. Otherwise, Stack-inside-Homebrew will download and compile
all dependencies on every CI job.

ttuegel committed:
[TMP] Jenkinsfile: Run macOS package on pull requests

ttuegel committed:
[TMP] Jenkinsfile: Check permissions on /var/cache

ttuegel commented:
Homebrew warns about using `HOMEBREW_NO_ENV_FILTERING`:

```
Warning: HOMEBREW_NO_ENV_FILTERING is undocumented, deprecated and will be removed in a future Homebrew release (because it breaks many things)!
```

Also, the Anka VM is missing the cache at `/opt/stack`.

ttuegel committed:
[TMP] Jenkinsfile: Create /opt/stack

Eventually, this will be created in the Anka template. For now, I just need to
test that the Homebrew builder can use it.

ttuegel committed:
[TMP] Jenkinsfile: Set STACK_ROOT inside workspace

ttuegel committed:
[TMP] Jenkinsfile: Dump environment

ttuegel committed:
[TMP] Jenkinsfile: Set STACK_ROOT=$HOME/.stack

ttuegel committed:
[TMP] Jenkinsfile: Enable Homebrew build again

ttuegel committed:
[TMP] Jenkinsfile: Try to build Stack cache in the job

ttuegel committed:
[TMP] Jenkinsfile: Change ownership of STACK_ROOT
